### PR TITLE
Zebra stripe table

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The Terra Table was updated so that consumers could specify whether zebra striping is implemented in the table.

**Why it was changed:**
The change was made as table implementations vary on the usage of zebra striping.  Moreover, the data grid components that leverage the table require zebra striping.

### Testing

This change was tested using:

- [X] WDIO
- [X] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [X] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9700 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/table/Examples/ZebraStripedTable.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import Table from 'terra-table';
+
+const gridDataJSON = {
```
Comment (kenk2) [inline packages/terra-framework-docs/src/terra-dev-site/doc/table/Examples/ZebraStripedTable.jsx:4]: Non-blocking nit: This is table but here and other places using this data this is referencing grid.

File: packages/terra-table/src/subcomponents/Row.jsx [hunk 2]
```diff
@@ -38,6 +38,11 @@ const propTypes = {
    */
   isSelected: PropTypes.bool,
 
+  /**
+   * Boolean indicating whether zebra striping is enabled
```
Comment (sdadn) [inline packages/terra-table/src/subcomponents/Row.jsx:42]:
```suggestion
   * Boolean indicating whether zebra striping is enabled.
```

**Review by sdadn — APPROVED** [review]
Small nit but otherwise looks good!

File: packages/terra-table/src/subcomponents/Row.jsx [hunk 3]
```diff
@@ -99,6 +105,7 @@ function Row(props) {
       className={cx('row', {
         selected: isSelected,
         selectable: hasRowSelection,
+        'striped-table-row': isTableStriped,
```
Comment (kenk2) [inline packages/terra-table/src/subcomponents/Row.jsx:108]: I think we could name this `striped`? This also feels awkward compared to the other class names that we use here.

File: packages/terra-table/src/subcomponents/Row.jsx [hunk 4]
```diff
@@ -81,6 +86,7 @@ function Row(props) {
     hasRowSelection,
     id,
     isSelected,
+    isTableStriped,
```
Comment (kenk2) [inline packages/terra-table/src/subcomponents/Row.jsx:89]: nit: Like my other comment, could we just name this `isStriped`? The naming feels weird compared to the other variable names.

**Review by kenk2 — APPROVED** [review]
Only a few tiny comments, but otherwise LGTM!

File: packages/terra-table/src/subcomponents/Row.jsx (hunk 4 above)
Comment (cm9361) [reply packages/terra-table/src/subcomponents/Row.jsx:89]: This naming convention is similar.  The row may or may not be striped.  It is the parent table that has striping enabled.  We have CSS to control this at the row component level though.

File: packages/terra-table/src/subcomponents/Row.jsx (hunk 3 above)
Comment (kenk2) [reply packages/terra-table/src/subcomponents/Row.jsx:108]: I think in that case, we can leave this as-is as was the intention that the table is the component that is passing this down to the component.

File: packages/terra-table/src/Table.jsx [hunk 5]
```diff
@@ -101,6 +101,11 @@ const propTypes = {
    * rendered to allow for row selection to occur.
    */
   hasSelectableRows: PropTypes.bool,
+
+  /**
+   * Boolean indicating whether or not the table should have zebra striping for rows.
```
Comment (eawww) [inline packages/terra-table/src/Table.jsx:106]:
```suggestion
   * Boolean specifying whether or not the table should have zebra striping for rows.
```
I know we use "indicating" in some places, but "specifying" or "defining" is more accurate.

File: packages/terra-table/src/subcomponents/Row.jsx [hunk 6]
```diff
@@ -38,6 +38,11 @@ const propTypes = {
    */
   isSelected: PropTypes.bool,
 
+  /**
+   * Boolean indicating whether zebra striping is enabled.
```
Comment (eawww) [inline packages/terra-table/src/subcomponents/Row.jsx:42]:
```suggestion
   * Boolean specifying whether zebra striping is enabled.
```

**Review by eawww — APPROVED** [review]
Lil nit but overall looks good

**Pull request merged**